### ISSUE TITLE
Rethink namespaces for the test suite

### ISSUE THREAD

**Issue by casparvl** [EESSI/test-suite#25]
Right now, are namespaces are a bit messy: `eessi_checks` and `eessi_utils` are installed as two separate packages, and form the top level namespaces. That means we get imports like this in the tests:

```
from eessi_utils import hooks
from eessi_utils import utils
```

Furthermore, if we ever inherit from some base mixin class, we'd get things like
```
from eessi_checks import eessi_testbase
```

All pretty ugly.

Discussed a bit with Sam on Slack. It might be nice to have everything related to testing in a `eessi.testing` namespace (we want to avoid `eessi.test` or `eessi.tests`, as we probably want to reserve the `.tests` for the _actual_ reframe test classes). E.g. we'd have things like `eessi.testing.utils`, `eessi.testing.hooks`, `eessi.testing.generic.eessi_baseline` (a baseline class for mixin inheritance), `eessi.testing.tests.tensorflow` (a parent class from which multiple TensorFlow tests could be derived, much like we derive our current GROMACS test from `hpctestlib.sciapps.gromacs.benchmarks`).

**Comment by smoors** [comment]
i agree it would be better to have a dedicated namespace for our tests, under a general `eessi` namespace.

just one suggestion: maybe we could just use `eessi.test_suite` instead of `eessi.testing` to better match this repo?

**Comment by boegel** [comment]
I'm not a big fan of the underscore either, but now we're really [bikeshedding](https://en.wiktionary.org/wiki/bikeshedding)...

Other options:
* `eessi.testsuite.*`
* `eessi.tests.*`

**Comment by boegel** [comment]
As discussed, let's settle on `eessi.testsuite.*`

That would give:
```
eessi.testsuite.utils
eessi.testsuite.tests.apps.gromacs
```
